### DISCUSSION TITLE
view action with slug instead of id number

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#12934] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.71

### How can we help you?

```
ViewAction::make()
  ->url(fn (Post $record): string => static::getUrl('view', ['record' => $record->slug]))
  ->openUrlInNewTab(),
```

to code below it open the url like this: `https://myapp.test/app/posts/what-is-lorem-ipsum
`
but I get `404 NOT FOUND` even that post slug already exist in database `slug` column

```
public static function getPages(): array
    {
        return [
            'index' => Pages\ListProducts::route('/'),
            'view' => Pages\ViewProduct::route('/{record}'),
        ];
    }
```

**Comment by dissto** [comment, marked as answer]
Maybe you can try

```php
public static function getRecordRouteKeyName(): ?string
{
    return 'slug';
}
```
on your resource class 🤔 

[Or even on your model](https://laravel.com/docs/11.x/routing#customizing-the-default-key-name)

**Reply by rezadonny** [reply]
Works Perfect!!
